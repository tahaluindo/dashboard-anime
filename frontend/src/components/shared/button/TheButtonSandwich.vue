<template>
  <div class="cursor-pointer mr-4" ref="button">
    <div class="h-1 w-7 mb-1 rounded bg-current"></div>
    <div class="h-1 w-7 mb-1 rounded bg-current"></div>
    <div class="h-1 w-7 rounded bg-current"></div>
  </div>
</template>

<script lang="ts">
import { defineComponent, onMounted, ref } from "vue";
import { AnimationController } from "../../../controllers/AnimationController";
import { AnimeSandwich } from "./controllers/AnimeSandwich";

export default defineComponent({
  setup() {
    const button = ref(null as unknown as HTMLElement);
    let animationController = new AnimationController<AnimeSandwich>();

    onMounted(() => {
      animationController.animation = new AnimeSandwich(button.value);
    });

    return {
      button,
      animationController,
    };
  },
});
</script>
