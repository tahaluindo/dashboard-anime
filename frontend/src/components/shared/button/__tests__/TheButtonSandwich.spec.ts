import { mount } from "@vue/test-utils";
import TheButtonSandwich from "../TheButtonSandwich.vue";

describe("TheButtonSandwich component test", () => {
  it("should render the component default state", () => {
    const wrapper = mount(TheButtonSandwich);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
