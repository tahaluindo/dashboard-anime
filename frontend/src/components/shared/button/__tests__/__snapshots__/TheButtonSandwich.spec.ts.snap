// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheButtonSandwich component test should render the component default state 1`] = `
"<div class=\\"cursor-pointer mr-4\\">
  <div class=\\"h-1 w-7 mb-1 rounded bg-current\\" style=\\"transform: translateY(8.881784197001252e-16px) ;\\"></div>
  <div class=\\"h-1 w-7 mb-1 rounded bg-current\\" style=\\"opacity: 0.9999999999999999; transform: translateY(0px) ;\\"></div>
  <div class=\\"h-1 w-7 rounded bg-current\\" style=\\"transform: translateY(-8.881784197001252e-16px) ;\\"></div>
</div>"
`;
